docarray/array/mixins/io/binary.py

```diff
@@ -22,17 +22,18 @@ def load_binary(
         protocol: str = 'pickle-array',
         compress: Optional[str] = None,
         _show_progress: bool = False,
+        return_iterator: bool = False,
     ) -> 'T':
         """Load array elements from a LZ4-compressed binary file.
 
         :param file: File or filename or serialized bytes where the data is stored.
         :param protocol: protocol to use
         :param compress: compress algorithm to use
         :param _show_progress: show progress bar, only works when protocol is `pickle` or `protobuf`
-
+        :param return_iterator: returns an iterator over the DocumentArray.
```

Comment:
what happens if protocol is another? why does protocol affect here?

Reply:
Depending on the procol loading changes (as it does in the `main` branch, the behaviour is kept the same)